### PR TITLE
refactor(lib): upgrade to Angular v6 and Material v6 betas

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Move to tree-shakeable providers where appropriate
* Deprecate unneeded providers
* Update tsconfig to workaround angular/angular#22210

### CODE REVIEW

File: test/karma-test-shim.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ System.config({
     '@angular/cdk/scrolling': 'node:@angular/cdk/bundles/cdk-scrolling.umd.js',
     '@angular/cdk/stepper': 'node:@angular/cdk/bundles/cdk-stepper.umd.js',
     '@angular/cdk/table': 'node:@angular/cdk/bundles/cdk-table.umd.js',
+    '@angular/cdk/tree': 'node:@angular/cdk/bundles/cdk-tree.umd.js',
```
Comment (ThomasBurleson) [inline test/karma-test-shim.js]: Why are we including the tree ? It is not used in our demos ?
> Then again neither is table ;-)

Comment (CaerusKaru) [reply test/karma-test-shim.js]: It's required for testing, otherwise Karma will try looking for it and then complain

**Comment by CaerusKaru** [comment]
Currently waiting for angular/angular#22655

**Comment by CaerusKaru** [comment]
Superseded by #681 to allow for separating out the tree-shakeable providers PR for later

**Pull request closed without merging**